Updated plugin source for VIP compatibility

bcampeau opened the pull request:
We're ready to submit this plugin to WordPress VIP for inclusion into their plugins repository. In order to do so, I've created this pull request which addresses all of the items likely to get your plugin approved. These include:

- Adding PHPDoc descriptions of all functions
- Fixing incorrect or missing output escaping
- Adding better translation compatibility
- Bringing code formatting up to WordPress PHP standards

I've also incremented the plugin version and added a few useful filters, though I can change that back if you don't wish to do so.

If this looks good, please merge this so I can get the VIP team reviewing it. Don't hesitate to follow up with any questions or comments. This fully addresses #30.

Thanks!

reefdog commented on documentcloud.php:
```diff
@@ -25,73 +25,104 @@
+		// Not needed on WordPress VIP since no other Documentcloud plugins exist.
+		if ( ! defined( 'WPCOM_IS_VIP_ENV' ) || ! WPCOM_IS_VIP_ENV ) {
+			add_action( 'admin_init', array( $this, 'check_dc_plugin_conflict') );
+	        add_action( 'init', array( $this, 'register_dc_oembed_provider' ) );
```
@bcampeau Sensible to only `check_dc_plugin_conflict` outside of WP VIP, but `register_dc_oembed_provider` always needs to run, so should be outside of this conditional block. Also, the indentation seems a bit off here?

reefdog commented on documentcloud.php:
```diff
@@ -25,73 +25,104 @@
+	 * Constructor.
+	 */
     function __construct() {
+		// Check for conflicts with other Documentcloud plugins.
```
@bcampeau Mind replacing `Documentcloud` with `DocumentCloud` wherever it appears in your comments?

reefdog commented on documentcloud.php:
```diff
@@ -25,73 +25,104 @@
         ?>
         <div class="error">
-            <p><?php _e( '<b>Warning!</b> You have two conflicting DocumentCloud plugins activated. Please deactivate Navis DocumentCloud, which has been replaced by <a target="_blank" href="https://wordpress.org/plugins/documentcloud/">DocumentCloud</a>.', 'documentcloud-plugin-conflict' ); ?></p>
+            <p><?php echo wp_kses( __( '<b>Warning!</b> You have two conflicting DocumentCloud plugins activated. Please deactivate Navis DocumentCloud, which has been replaced by <a target="_blank" href="https://wordpress.org/plugins/documentcloud/">DocumentCloud</a>.', 'documentcloud' ) ); ?></p>
```
@bcampeau I wasn't familiar with `wp_kses()`, but my reading of its [docs](https://codex.wordpress.org/Function_Reference/wp_kses) indicate we need to pass in a whitelist of HTML tags. Does it work fine without?

reefdog commented on documentcloud.php:
```diff
@@ -25,73 +25,104 @@
+			add_filter( 'http_request_host_is_external', '__return_true');
+		*/
+
+        $oembed_resource_domain = apply_filters( 'documentcloud_oembed_resource_domain', WP_DocumentCloud::OEMBED_RESOURCE_DOMAIN );
```
@bcampeau Not familiar with the purpose of `apply_filters()` used here, could you fill me in?

bcampeau replied:
This is intended so people can update this within their theme if needed for testing rather than comment it in the plugin code. It's more of a WordPress way to do. Also if they were testing on a VIP preprod site for instance, editing this code would not be an option.

reefdog commented on documentcloud.php:
```diff
@@ -262,120 +323,156 @@ function clean_dc_url($url) {
+     */
+    function render_options_page() {
+    	?>
+        <h2><?php esc_html_e( 'DocumentCloud Options', 'documentcloud' ) ?></h2>
```
@bcampeau Does this need to be a `esc_html_e()` instead of just `__()`?

reefdog commented on documentcloud.php:
```diff
@@ -262,120 +323,156 @@ function clean_dc_url($url) {
+        		'',
+                '__return_null',
+                'documentcloud'
+            );
```
@bcampeau Indentation on L358-363 seems a little weird, is it correct?

reefdog commented on documentcloud.php:
```diff
@@ -262,120 +323,156 @@ function clean_dc_url($url) {
+     */
+    function save( $post_id, $post ) {
+    	// Avoid autosave
+		if ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE ) {
```
@bcampeau Odd indentation?

reefdog commented:
@bcampeau As you can see, I added a bunch of line notes with individual questions or quibbles. Overall this looks great and I'll be glad to merge it in.

bcampeau commented:
I've addressed all your comments. There was some very weird indentation issue with the file. I had to completely zap your existing tabs but all is good now.

bcampeau commented:
Just wanted to check in about getting this merged so we can proceed with a VIP review.

Thanks.

reefdog commented on documentcloud.php:
```diff
@@ -3,379 +3,476 @@
+		// Not needed on WordPress VIP since no other DocumentCloud plugins exist.
+		if ( ! defined( 'WPCOM_IS_VIP_ENV' ) || ! WPCOM_IS_VIP_ENV ) {
+			add_action( 'admin_init', array( $this, 'check_dc_plugin_conflict') );
+			add_action( 'init', array( $this, 'register_dc_oembed_provider' ) );
```
@bcampeau We'll still need to register DC as an oEmbed provider, so this line needs to be outside the conditional block. Unless VIP also means we're whitelisted as an oEmbed provider, which would be *wonderful*.

bcampeau replied:
Yeah, this was just a mistake. I'll fix now.

reefdog commented:
@bcampeau Sorry, dragging a bit on testing this. I'll try to have it merged in today, else tomorrow. Thanks again for your work on this. Added one note above.

bcampeau commented:
Thanks. I missed that comment on the oEmbed and I've just fixed it now.

reefdog commented:
Super. I'll give it a test today and merge in unless I find problems. Thanks again @bcampeau.

bcampeau commented:
Any issues in testing? Thanks.

reefdog commented:
@bcampeau No problems from your plugin, though I found a few other things I want to fix. Thanks again, shooting for a release this week.

bcampeau commented:
Thanks. I've given VIP a heads up ahead of time since it will take some time for them to get this in their queue.
